### ISSUE TITLE
Re-add a robust EOT-head learning test (dropped flaky one)

### ISSUE THREAD

**Issue by beyarkay** [beyarkay/factorion#172]
`TestEotHead::test_eot_head_learns_terminal_vs_placement` was **removed** because it's flaky — it asserts a noise-level result. We still want a test of the underlying contract (the EOT head learns to fire on completed factories so a rollout can "run until it thinks it's done"), so this issue tracks re-adding a **robust** version.

### What the dropped test did
Trained a tiny SFT model (`size=5, num_samples=1000, max_level=3, epochs=20`, hardcoded `seed=3`), then on 40 held-out `MOVE_ONE_ITEM` factories asserted `pos_mean > neg_mean` — the eot-head's mean logit on *solved* factories exceeds its mean on *partial* (3-removed) ones.

### Why it's flaky
At that scale the eot head barely separates the two classes, so the bare inequality reads noise. Sweeping the training seed 0–19 (same logic):

```
margins (pos_mean - neg_mean), all on logits around -1.5:
 +0.003 -0.034 +0.064 -0.021 -0.017 +0.112 +0.001 +0.136 +0.065 +0.024
 +0.054 +0.023 +0.034 -0.057 -0.028 +0.040 +0.058 +0.038 +0.038 +0.010
=> 15/20 PASS (~75%); every margin within ±0.14, most within ±0.05
```

So pass/fail is ~a coin flip per seed. It surfaced when an unrelated throughput/connectivity change (the underground-belt connection rewrite) nudged `seed=3`'s already-near-zero margin from + to − — the test broke despite the change not touching EOT learning at all.

### What a robust replacement should do
Pick one (or combine):
- **Demand a real margin** (`pos_mean - neg_mean > MARGIN`) with a model big enough / trained long enough that the separation is genuine, not ~0.
- **Average over several seeds** and assert the mean separation, or that a high fraction (e.g. ≥18/20) separate — not a single seed's sign.
- **Assert val EOT accuracy / positive-recall above a threshold** using the per-`LessonKind` eot metrics `train_sft` already computes (`val/<LESSON>/eot_acc`, `eot_pos_recall`), which is a more direct measure than raw logit means.

Context: dropped as part of the underground-belt connectivity work.
